recipes/dcmtk/all/conanfile.py

```diff
@@ -51,149 +52,190 @@ class DCMTKConan(ConanFile):
+        tc.variables["DCMTK_WITH_DOXYGEN"] = False
+        tc.variables["DCMTK_WIDE_CHAR_FILE_IO_FUNCTIONS"] = self.options.wide_io
+        tc.variables["DCMTK_WIDE_CHAR_MAIN_FUNCTION"] = self.options.wide_io
+        tc.variables["DCMTK_ENABLE_STL"] = self.options.enable_stl
+        tc.variables["DCMTK_ENABLE_CXX11"] = True
+        tc.variables["DCMTK_ENABLE_MANPAGE"] = False
+        tc.cache_variables["DCMTK_DEFAULT_DICT"] = self.options.default_dict
+        tc.variables["DCMTK_USE_DCMDICTPATH"] = self.options.use_dcmdictpath
         if self.settings.os == "Windows":
-            cmake.definitions["DCMTK_OVERWRITE_WIN32_COMPILER_FLAGS"] = False
-
+            tc.variables["DCMTK_OVERWRITE_WIN32_COMPILER_FLAGS"] = False
         if is_msvc(self):
-            cmake.definitions["DCMTK_ICONV_FLAGS_ANALYZED"] = True
-            cmake.definitions["DCMTK_COMPILE_WIN32_MULTITHREADED_DLL"] = "MD" in msvc_runtime_flag(self)
+            tc.variables["DCMTK_ICONV_FLAGS_ANALYZED"] = True
+            tc.variables["DCMTK_COMPILE_WIN32_MULTITHREADED_DLL"] = not is_msvc_static_runtime(self)
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def _collect_cmake_required(self, dependency):
+        dep = self.dependencies.host[dependency]
+        dep_cpp_info = dep.cpp_info.aggregated_components()
+        dep_includes = [p.replace("\\", "/") for p in dep_cpp_info.includedirs]
+        dep_defs = [d for d in dep_cpp_info.defines]
+        libdirs_flag = "/LIBPATH:" if is_msvc(self) else "-L"
+        dep_libdirs = ["{}{}".format(libdirs_flag, p.replace("\\", "/")) for p in dep_cpp_info.libdirs]
+        dep_libs = [l for l in dep_cpp_info.libs]
+        dep_system_libs = [s for s in dep_cpp_info.system_libs]
+        dep_frameworks = [f"-framework {f}" for f in dep_cpp_info.frameworks]
+        for _, trans_dependency in dep.dependencies.items():
+            if trans_dependency.context != "host":
+                continue
+            trans_dep_cpp_info = trans_dependency.cpp_info.aggregated_components()
+            dep_includes.extend([p.replace("\\", "/") for p in trans_dep_cpp_info.includedirs])
+            dep_defs.extend([d for d in trans_dep_cpp_info.defines])
+            dep_libdirs.extend(["{}{}".format(libdirs_flag, p.replace("\\", "/")) for p in trans_dep_cpp_info.libdirs])
+            dep_libs.extend([l for l in trans_dep_cpp_info.libs])
+            dep_system_libs.extend([s for s in trans_dep_cpp_info.system_libs])
+            dep_frameworks.extend([f"-framework {f}" for f in trans_dep_cpp_info.frameworks])
 
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+        return {
+            "includes": ";".join(dep_includes),
+            "definitions": ";".join(dep_defs),
+            "link_options": ";".join(dep_libdirs + dep_frameworks),
+            "libraries": ";".join(dep_libs + dep_system_libs),
+        }
```

Comment:
It works fine, and I don't want to invalidate reviews and trigger a new heavy build, but next time I'll try this:

```python
    def _collect_cmake_required(self, dependency):
        """
        Collect all compile & link flags of a dependency so that they can be used in:
          - CMAKE_REQUIRED_FLAGS
          - CMAKE_REQUIRED_INCLUDES
          - CMAKE_REQUIRED_DEFINITIONS
          - CMAKE_REQUIRED_LINK_OPTIONS
          - CMAKE_REQUIRED_LIBRARIES
        These variables are listened by CMake functions like check_symbol_exists() etc
        """
        # Aggregate cpp_info of dependency and all its depencencies (direct and transitive)
        dep = self.dependencies.host[dependency]
        dep_cpp_info = dep.cpp_info.aggregated_components()
        for _, trans_dep in dep.dependencies.items():
            if trans_dep.context == "host":
                dep_cpp_info.merge(trans_dep.cpp_info.aggregated_components())

        # Get compile & link flags of dependency and its dependencies
        dep_cflags = [cf for cf in dep_cpp_info.cflags]
        dep_includes = [p.replace("\\", "/") for p in dep_cpp_info.includedirs]
        dep_defs = [d for d in dep_cpp_info.defines]
        libdirs_flag = "/LIBPATH:" if is_msvc(self) else "-L"
        dep_libdirs = ["{}{}".format(libdirs_flag, p.replace("\\", "/")) for p in dep_cpp_info.libdirs]
        dep_linkflags = [lf for lf in dep_cpp_info.linkflags]
        dep_frameworks = [f"-framework {f}" for f in dep_cpp_info.frameworks]
        dep_libs = [l for l in dep_cpp_info.libs]
        dep_system_libs = [s for s in dep_cpp_info.system_libs]

        return {
            "flags": ";".join(dep_cflags),
            "includes": ";".join(dep_includes),
            "definitions": ";".join(dep_defs),
            "link_options": ";".join(dep_libdirs + dep_linkflags + dep_frameworks),
            "libraries": ";".join(dep_libs + dep_system_libs),
        }
```

(inspired by https://docs.conan.io/2/reference/tools/cpp_info.html#aggregating-information-in-custom-generators)